llvm/lib/Target/RISCV/Disassembler/RISCVDisassembler.cpp

```diff
@@ -367,6 +367,18 @@ DecodeStatus RISCVDisassembler::getInstruction(MCInst &MI, uint64_t &Size,
     }
     Insn = support::endian::read16le(Bytes.data());
 
+    if (!STI.getFeatureBits()[RISCV::FeatureExtXCoreVAlu]) {
+      LLVM_DEBUG(
+          dbgs() << "Trying COREV_ALU32 table (32-bit Instruction):\n");
+      // Calling the auto-generated decoder function.
+      Result = decodeInstruction(DecoderTableCOREV_ALU32, MI, Insn, Address,
+                                 this, STI);
+      if (Result != MCDisassembler::Fail) {
+        Size = 4;
+        return Result;
+      }
+    }
```

Comment:
IIUC this only deals with the `cv.ror` command? I don't quite get, why this is necessary. AFAICT no CORE-V instructions can even reach this code, since all of the instructions have an opcode ending in `0b11`, since cv32e40p is a 32-bit platform.

Reply:
Yes, I must have made a mistake here.

The different decoding namespace is necessary, because otherwise the disassembler creates an `<unknown>` when there is an ambiguous instruction encoding, making some tests on the disassembler fail.

Anyway, this shouldn't matter, since `cv.ror` isn't supported anymore.